DuckDuckGo/SettingsViewController.swift

```diff
@@ -355,22 +356,30 @@ class SettingsViewController: UITableViewController {
     private func configureNetPCell() {
         netPCell.isHidden = !shouldShowNetPCell
 #if NETWORK_PROTECTION
+        updateNetPCellSubtitle(connectionStatus: connectionObserver.recentValue)
         connectionObserver.publisher
             .receive(on: DispatchQueue.main)
             .sink { [weak self] status in
-                let detailText: String
-                switch status {
-                case .connected:
-                    detailText = UserText.netPCellConnected
-                default:
-                    detailText = UserText.netPCellDisconnected
-                }
-                self?.netPCell.detailTextLabel?.text = detailText
+                self?.updateNetPCellSubtitle(connectionStatus: status)
             }
             .store(in: &cancellables)
 #endif
     }
 
+#if NETWORK_PROTECTION
+    private func updateNetPCellSubtitle(connectionStatus: ConnectionStatus) {
```

Comment:
@diegoreymendez Gave me feedback a while back to extract this to a ViewModel. I’d like to add a tech debt task to do that to the backlog to come back to later. Don’t think it would take too long.

Reply:
Yeah agreed, I'd imagine a full `SettingsViewModel` which would handle (and allow testing of) the logic we have for showing/hiding cells and their copy values.